### PR TITLE
gh-107298: Add standard exceptions and warnings in the nitpick_ignore list

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
* Issue: gh-107298

----
:books: Documentation preview :books:: https://cpython-previews--108029.org.readthedocs.build/

### CODE REVIEW

File: Doc/conf.py [hunk 1]
```diff
@@ -157,6 +157,77 @@
     ('envvar', 'USER'),
     ('envvar', 'USERNAME'),
     ('envvar', 'USERPROFILE'),
+]
+
+# Temporary undocumented names.
+# In future this list must be empty.
+nitpick_ignore = [
```
Comment (AA-Turner) [inline Doc/conf.py]: Currently this overwites the list above:
```suggestion
```

A

Comment (serhiy-storchaka) [reply Doc/conf.py]: Oh, I was going to use `+=`.

**Review by AA-Turner — APPROVED** [review]

File: Doc/conf.py [hunk 2]
```diff
@@ -157,6 +157,77 @@
     ('envvar', 'USER'),
     ('envvar', 'USERNAME'),
     ('envvar', 'USERPROFILE'),
+]
+
+# Temporary undocumented names.
+# In future this list must be empty.
```
Comment (vstinner) [inline Doc/conf.py:163]: A fix would be to replace ``:c:data:`Exception` `` with ``.. c:data:`Exception` `` in Doc/c-api/exceptions.rst, right?

Comment (serhiy-storchaka) [reply Doc/conf.py:163]: I tried this, and it turned out not so simple. You need to specify the type:
```
.. c:var:: PyObject *PyExc_SomeError
```
and the result does not look good.

**Pull request merged**